pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -258,16 +258,23 @@ def restore_loops(self) -> None:
 
     def restore_optimizers_and_schedulers(self) -> None:
         """Restores the optimizers and learning rate scheduler states from the pre-loaded checkpoint."""
-        if not self._loaded_checkpoint or not self.trainer.strategy.lightning_restore_optimizer_and_schedulers:
```

Comment:
The fix is here. We make sure we let lightning restore the lr scheduler regardless of whether deepspeed manages the stepping or not. We still let the strategy decide whether it wants to let lightning restore the optimizer states or not. Since we are separating this logic, I decided to also split the error message in two.